UN-OCHA/hid-api
# HID-2259: clean up UserController.create

# HID-2259

A general refactor to meet the following goals:

- Make the code linear and understandable.
- Validate the input more strictly, and provide more consistent feedback when returning HTTP 400s.
- Remove legacy attributes and legacy workarounds for long-past problems.

## Testing

- `doco exec api npm i && doco restart api`
- Attempt to create users via API call and look for 200/400 responses. If you can produce a 500, this PR needs work.
- Try anything you want: malformed emails, missing props, additional props in the object, go nuts!
- Use the website to perform a password reset and ensure the password dictionary still works. I had to refactor the function from a mongoose "method" to a "static" so that it could be run on a not-yet-created user object.
